pyiron_atomistics/atomistics/structure/analyse.py

```diff
@@ -47,6 +76,234 @@ def get_average_of_unique_labels(labels, values):
+            use_voronoi (bool): Use Voronoi vertices for the initial interstitial candidate
+                positions instead of grid points.
+        """
+        self.min_distance = min_distance
+        self.structure = structure
+        if use_voronoi:
+            self.positions = structure.analyse.get_voronoi_vertices()
+        else:
+            self.positions = self._create_gridpoints(
+                n_gridpoints_per_angstrom=n_gridpoints_per_angstrom
+            )
+        if min_distance > 0:
+            self._remove_too_close()
+
+    @property
+    def num_neighbors(self):
+        return self._num_neighbors
+
+    @num_neighbors.setter
+    def num_neighbors(self, n):
+        self.reset()
+        self._num_neighbors = n
+
+    def reset(self):
+        self._hull = None
+        self._neigh = None
+
+    @property
+    def neigh(self):
+        if self._neigh is None:
+            self._neigh = self.structure.get_neighborhood(
+                self.positions, num_neighbors=self.num_neighbors
+            )
+        return self._neigh
+
+    @property
+    def positions(self):
+        return self._positions
+
+    @positions.setter
+    def positions(self, x):
+        self.reset()
+        self._positions = x
+
+    @property
+    def hull(self):
+        if self._hull is None:
+            self._hull = [ConvexHull(v) for v in self.neigh.vecs]
+        return self._hull
+
+    def _create_gridpoints(self, n_gridpoints_per_angstrom=5):
+        cell = self.structure.cell.diagonal()
+        gridpoints = (n_gridpoints_per_angstrom*cell).astype(int)
+        positions = [np.linspace(0, cell[i], gridpoints[i], endpoint=False) for i in range(3)]
+        positions = np.meshgrid(*positions)
+        return np.stack(positions, axis=-1).reshape(-1, 3)
+
+    def _remove_too_close(self):
+        neigh = self.structure.get_neighborhood(self.positions, num_neighbors=1)
+        self.positions = self.positions[neigh.distances.flatten()>self.min_distance]
+
+    def _set_interstitials_to_high_symmetry_points(self):
+        self.positions = self.positions+np.mean(self.neigh.vecs, axis=-2)
+        self.positions = self.structure.get_wrapped_coordinates(self.positions)
+
+    def _kick_out_points(self, variance_buffer=0.01):
+        variance = self.get_variance()
+        min_var = variance.min()
+        self.positions = self.positions[variance<min_var+variance_buffer]
+
+    def _cluster_points(self, eps=0.1):
+        extended_positions, indices = self.structure.get_extended_positions(
+            eps, return_indices=True, positions=self.positions
+        )
+        labels = DBSCAN(eps=eps, min_samples=1).fit_predict(extended_positions)
+        coo = coo_matrix((labels, (np.arange(len(labels)), indices)))
+        labels = coo.max(axis=0).toarray().flatten()
+        self.positions = get_mean_positions(
+            self.positions, self.structure.cell, self.structure.pbc, labels
+        )
+
+    def get_interstitials(
```

Comment:
UI-wise `interstitials.get_interstitials()` feels a bit clunky, what about

```python
... = structure.analyse.iterstitials.get_positions()
```

or 

```python
structure.analyse.interstitials.calculate()
... = structure.analyse.interstitials.positions
```

Reply:
Purely cosmetic, though, I won't insist.

Reply:
I like the suggestion